### PR TITLE
Add `llvm-skip-rebuild` flag to `x.py`

### PR DESCRIPTION
**Pull request by matthew-healy**
This PR follows on from #67437 to complete the feature request from #65612.

Specifically it adds a new command-line flag, `--llvm-skip-rebuild`, which overrides both any value set in `config.toml` and the default value (`false`).

I'm not 100% confident that I've implemented the override in the "best" way, but I've checked it locally and it seems to work at least.

This option isn't currently mentioned in the Guide to Rustc Development. I'd be happy to write something on it if folk think that's worthwhile.

### CODE REVIEW

File: src/bootstrap/flags.rs [hunk 1]
```diff
@@ -487,6 +497,9 @@ Arguments:
                 .map(|p| p.into())
                 .collect::<Vec<_>>(),
             deny_warnings: parse_deny_warnings(&matches),
+            llvm_skip_rebuild: matches
+                .opt_str("llvm-skip-rebuild")
+                .and_then(|s| s.parse::<bool>().ok()),
```
Comment (Centril) [inline src/bootstrap/flags.rs]:
```suggestion
                .map(|s| s.parse::<bool>().expect("`llvm-skip-rebuild` should be true/false")),
```

**Comment by Centril** [comment]
r? @Centril 

r=me with comment ^-- addressed.

**Comment by matthew-healy** [comment]
@Centril Thanks for the review! I've updated the PR. I also added a `to_lowercase()` so that users don't have to worry about the casing requirements for correctly parsing a `bool`. Happy to remove that if you'd rather not have it, though.

**Comment by Centril** [comment]
It's not something we typically do, but it also seems fine to me. :)

@bors r+

**Pull request merged**